test/test_integration_diagnostics.cxx

```diff
@@ -441,3 +444,59 @@ TEST_CASE("integration: ping allows to select bucket and opens it automatically"
         REQUIRE(res.services[couchbase::core::service_type::key_value][0].bucket.value() == integration.ctx.bucket);
     }
 }
+
+TEST_CASE("integration: ping allows setting timeout", "[integration]")
```

Comment:
I think this test might have to be removed - it checks that with 1 millisecond timeout the ping for all services times out, which might not necessarily be the case. Do we have a good way to test that timeouts work?

Reply:
No, I think it makes sense. Do think it would be source of unstable tests when sometimes the ping will not timeout in one millisecond?

Reply:
Yes, potentially, if it manages to ping before the timeout. So far that does not seem to be the case either on localhost or CI so we could just keep an eye on it and remove it if that's an issue

Reply:
OK